src/server/controllers/user.js

```diff
@@ -20,12 +20,37 @@ const postRegisterDetails = async (req, res) => {
         res.json({newUser: newUser});
         
     } catch(error) {
-        res.status(500).send("Something Broke");
+        res.status(500).send("There was an error!");
     }
 
 
 }
 
+const postLoginDetails = async (req, res) => {
+    try {
+        const { username, password } = req.body;
+        const findUser = await prisma.user.findUnique({
+            where: {
+                username
+            }
+        });
+        if (findUser) {
+            if (await bcrypt.compare(password, findUser.password)) {
+                const token = jwt.sign(username, secret);
+                return res.json({ userToken: token })
+            } else {
+                return res.json("Username or Password is incorrect.")
+            }
+        } else {
+            return res.json("Username or Password is incorrect.");
+        }
```

Comment:
perhaps there is a way to combine both of these `else { return }` into a single else, and combine findUser and bcrypt.compare into the one condition? Regardless, this reads well and works efficiently.

Reply:
WIll it work if we just have "return" only on line 42 instead of that whole line thats currently there?

Reply:
Ok. After we merge this let's have a mini test session with this block of code. I'm going to have to see it in action.

Reply:
you could also combine the two if statements into an AND, but it might be a bit long on a single line, Steve/Nathan what do you think about nested IFs vs a single long one?